gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/HttpClientServiceImpl.java

```diff
@@ -237,6 +245,7 @@ public void handle(Throwable throwable) {
         try {
             return promise.future().toCompletionStage().toCompletableFuture().get();
         } catch (InterruptedException | ExecutionException e) {
+            Thread.currentThread().interrupt();
```

Comment:
I think we should not interrupt the thread in case of an `ExecutionException` and rather split the catch clause in two